### PR TITLE
Add integration test for NodeResourcesFit in requeueing scenarios

### LINKED ISSUE #122305: Implement the integration tests for requeueing scenarios
/sig scheduling
/priority important-soon
/kind feature

Part of: https://github.com/kubernetes/kubernetes/issues/122284#issuecomment-1853346333.

We have few tests to check the scheduler's requeueing scenario in the integration test. 
We should have the one so that we can catch a bug like https://github.com/kubernetes/kubernetes/issues/122284, which only happens the scheduling queue is actually working.

---

Here's the list of plugins with QueueingHint.
Basically, I believe we can just pile up the test cases in the existing [TestCoreResourceEnqueue](https://github.com/kubernetes/kubernetes/blob/746f08a8da0a45e8b1a46362822db05cb11ed294/test/integration/scheduler/queue_test.go#L188).
Also, note that we have to cover major scenarios only, don't have to cover minor ones within the integration test.

**Please comment in this issue before starting to work on so that we can avoid many people from working on the same thing!**

- [x] InterPodAffinity @nayihz https://github.com/kubernetes/kubernetes/pull/127090
- [x] Dynamic resource allocation https://github.com/kubernetes/kubernetes/issues/122305#issuecomment-2336618368
- [x] NodeAffinity @tozastation https://github.com/kubernetes/kubernetes/pull/127139
- [x] NodePorts @saku3 https://github.com/kubernetes/kubernetes/pull/127130
- [x] NodeResourceFit @unvavo https://github.com/kubernetes/kubernetes/pull/127354
- [x] NodeUnschedulable @toshikish  https://github.com/kubernetes/kubernetes/pull/127406
- [ ] CSILimits @utam0k https://github.com/kubernetes/kubernetes/pull/127224
- [x] TopologySpread @saku3 https://github.com/kubernetes/kubernetes/pull/127399
- [x] TaintToleration @unvavo  https://github.com/kubernetes/kubernetes/pull/127923
- [ ] VolumeBinding @saku3  https://github.com/kubernetes/kubernetes/pull/127946
- [x] VolumeRestriction @utam0k https://github.com/kubernetes/kubernetes/pull/127892
- [x] VolumeZone @Gekko0114 https://github.com/kubernetes/kubernetes/pull/127353

### PR DESCRIPTION
**Pull request by unvavo**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Implement integration test for NodeResourcesFit in requeueing scenarios

#### Which issue(s) this PR fixes:

Fixes #122305 

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: test/integration/scheduler/queue_test.go [hunk 1]
```diff
@@ -295,7 +295,7 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			wantRequeuedPods: sets.New("pod1"),
 		},
 		{
-			name:         "Pod got resource scaled down requeued to activeQ",
+			name:         "Pod rejected by the NodeResourcesFit plugin is requeued when the Pod is updated to scale down",
```
Comment (unvavo) [inline test/integration/scheduler/queue_test.go:298]: I was going to add a case that a Pod resources are scaled down, but it already existed here.
I modified the name to make it clearer that it is a test for NodeResourcesFit.

**Comment by unvavo** [comment]
Could you review this PR when you have time?

/assign @macsko
for the first path.
/assign @sanposhiho
for the final approval

File: test/integration/scheduler/queue_test.go [hunk 2]
```diff
@@ -311,6 +311,24 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			},
 			wantRequeuedPods: sets.New("pod1"),
 		},
+		{
+			name:         "Pod rejected by the NodeResourcesFit plugin is requeued when a Node is updated",
```
Comment (macsko) [inline test/integration/scheduler/queue_test.go:315]: Let's add a second pod to this test case, that won't be requeued after the node is updated, e.g. still having too high requests.

Comment (macsko) [reply test/integration/scheduler/queue_test.go:315]: And could you also cover node add and pod delete events in a yet another test cases?

Comment (unvavo) [reply test/integration/scheduler/queue_test.go:355]: Thank you for reviewing.
I added a second pod that is not requeued.
I also added two other test cases to cover Node add and Pod delete events.
Could you check this, please?

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

**Pull request merged**